### ISSUE TITLE
Crashpad submodule reference doesn't point to a valid commit

### ISSUE THREAD

**Issue by csk-ableton** [getsentry/sentry-native#999]
**Description**

The Crashpad submodule reference doesn't point to a commit on a branch (should be `getsentry` I guess), see the error when visiting https://github.com/getsentry/crashpad/tree/f4c7710dc178218e1e58d9e0b65be91cc5dbd66c. I'm not sure how this commit is kept alive and I'm not able to checkout the commit. It is however possible to initialize the submodule fine from within this repository (I'm not clear why). 

I have a more complex integration of the library and manually checkout https://github.com/getsentry/crashpad which is currently not possible.

**When does the problem happen**

- [x] During build
- [ ] During run-time
- [ ] When capturing a hard crash

**Environment**

- OS: macOS

**Steps To Reproduce**

```
git clone https://github.com/getsentry/crashpad
cd crashpad
git checkout f4c7710dc178218e1e58d9e0b65be91cc5dbd66c
```

**Log output**

```
fatal: reference is not a tree: f4c7710dc178218e1e58d9e0b65be91cc5dbd66c
```

**Comment by csk-ableton** [comment]
Hi @supervacuus, thanks for the fast and detailed reply! I didn't realize the orphaned commit isn't fetched automatically and I have to do it explicitly (and that's what the submodule init is doing). That makes all sense then.

**Comment by supervacuus** [comment]
Fixed in #1000.
